services/email.py

```
@@ -0,0 +1,81 @@
+                email_data = {
+                    'id': message['id'],
+                    'from': next((h['value'] for h in headers if h['name'] == 'From'), ''),
+                    'subject': next((h['value'] for h in headers if h['name'] == 'Subject'), ''),
+                    'date': next((h['value'] for h in headers if h['name'] == 'Date'), ''),
+                    'snippet': msg.get('snippet', '')
+                }
+                emails.append(email_data)
+            
+            return emails
+        except Exception as e:
+            raise Exception(f"Error fetching emails: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)